File: go-backend/internal/domain/review/review.go
package review

import (
	"tanzanite/internal/domain/product"
	"tanzanite/internal/domain/user"
	"time"

	"gorm.io/gorm"
)

// Review 商品评价
type Review struct {
	ID           uint             `gorm:"primarykey" json:"id"`
	ProductID    uint             `gorm:"not null;index" json:"product_id"`
	UserID       uint             `gorm:"not null;index" json:"user_id"`
	OrderID      uint             `gorm:"index" json:"order_id"`
	Product      *product.Product `gorm:"foreignKey:ProductID" json:"product,omitempty"`
	User         *user.User       `gorm:"foreignKey:UserID" json:"-"`
	Rating       int              `gorm:"not null" json:"rating"` // 1-5星
	Title        string           `json:"title"`
	Content      string           `gorm:"type:text" json:"content"`
	Images       string           `gorm:"type:text" json:"images"`               // JSON数组
	Pros         string           `gorm:"type:text" json:"pros"`                 // 优点
	Cons         string           `gorm:"type:text" json:"cons"`                 // 缺点
	Status       string           `gorm:"index;default:'pending'" json:"status"` // pending, approved, rejected
	Featured     bool             `gorm:"default:false" json:"featured"`         // 是否精选
	Verified     bool             `gorm:"default:false" json:"verified"`         // 是否已购买验证
	HelpfulCount int              `gorm:"default:0" json:"helpful_count"`        // 有用数
	ReplyContent string           `gorm:"type:text" json:"reply_content"`        // 商家回复
	RepliedAt    *time.Time       `json:"replied_at"`
	RepliedBy    uint             `json:"replied_by"`
	CreatedAt    time.Time        `json:"created_at"`
	UpdatedAt    time.Time        `json:"updated_at"`
	DeletedAt    gorm.DeletedAt   `gorm:"index" json:"-"`
}

// TableName 指定表名
func (Review) TableName() string {
	return "reviews"
}

File: go-backend/web/admin/src/api/review.ts
import axios from '@/utils/axios'
import {
  requireApiArray,
  requireApiObject,
  requireApiPagination,
  unwrapApiPayload,
} from '@/utils/apiResponse'

export type ReviewStatus = 'pending' | 'approved' | 'rejected'
export type ReviewStatusFilter = ReviewStatus | 'all'

export interface AdminReviewProduct {
  id: number
  name: string
  sku: string
}

export interface AdminReviewUser {
  id: number
  username: string
  email: string
  display_name: string
}

export interface AdminReview {
  id: number
  product?: AdminReviewProduct | null
  user?: AdminReviewUser | null
  order_id: number
  rating: number
  title: string
  content: string
  images: string[]
  pros: string
  cons: string
  status: ReviewStatus
  featured: boolean
  verified: boolean
  helpful_count: number
  reply_content: string
  replied_at?: string | null
  replied_by: number
  moderated_at?: string | null
  moderated_by?: number | null
  moderation_reason: string
  created_at: string
  updated_at: string
}

export interface AdminReviewPagination {
  page: number
  page_size: number
  total: number
  total_pages: number
}

export interface AdminReviewListResult {
  data: AdminReview[]
  pagination: AdminReviewPagination
}

const readPaged = (response: unknown, path: string): AdminReviewListResult => {
  const responseBody = requireApiObject((response as { data?: unknown }).data, path, 'response body')
  const payload = unwrapApiPayload(response, path)
  return {
    data: requireApiArray<AdminReview>(payload, path, 'data'),
    pagination: requireApiPagination(responseBody, payload, path),
  }
}

const readObjectPayload = (response: unknown, path: string): AdminReview => (
  requireApiObject(unwrapApiPayload(response, path), path) as AdminReview
)

export const reviewApi = {
  async list(params: {
    status: ReviewStatusFilter
    search?: string
    product_id?: number
    page: number
    page_size: number
  }): Promise<AdminReviewListResult> {
    const path = '/api/admin/reviews'
    return readPaged(await axios.get(path, { params }), path)
  },

  async get(id: number): Promise<AdminReview> {
    const path = `/api/admin/reviews/${id}`
    return readObjectPayload(await axios.get(path), path)
  },

  async updateStatus(id: number, status: ReviewStatus, reason = ''): Promise<AdminReview> {
    const path = `/api/admin/reviews/${id}/status`
    return readObjectPayload(await axios.patch(path, { status, reason }), path)
  },
}

export default reviewApi
